src/main/java/spoon/support/reflect/declaration/CtElementImpl.java

```diff
@@ -127,8 +127,8 @@ public boolean equals(Object o) {
 			return false;
 		String current = getSignature();
 		String other = ((CtElement) o).getSignature();
-		if (current.length() <= 0 || other.length() <= 0)
-			throw new CtUncomparableException("Unable to compare elements");
+		//if (current.length() <= 0 || other.length() <= 0)
+		//	throw new CtUncomparableException("Unable to compare elements");
```

Comment:
Yes, that is the point. I removed the "IF + throw"  because that check does not allow to compare elements with empty signature. As consequence, if we have an empty CtLiteral (imagine as example the: code "".equals(myString) ) , we can not compare this element. I wrote a test case that shows this problem: if you remove the comment (i.e., keep the precondition) the comparison of an empty ctelement will fail. So, the patch I propose is to remove the if and throw statement